test/test_other.py

```diff
@@ -12226,6 +12226,28 @@ def test_asan_strncpy(self):
     # https://github.com/emscripten-core/emscripten/issues/14618
     self.do_runf('other/test_asan_strncpy.c', emcc_args=['-fsanitize=address'])
 
+  @parameterized({
+    'asan': ['AddressSanitizer: null-pointer-dereference on address 0x00000000', '-fsanitize=address'],
+    'safe_heap': ['Aborted(segmentation fault storing 4 bytes to address 0)', '-sSAFE_HEAP'],
+  })
+  @parameterized({
+    '': [],
+    'memgrowth': ['-pthread', '-sALLOW_MEMORY_GROWTH', '-Wno-pthreads-mem-growth'],
+  })
+  @parameterized({
+    '': [],
+    'bigendian': ['-sSUPPORT_BIG_ENDIAN'],
+  })
+  def test_null_deref_via_js(self, expected_output, *args):
+    # Multiple JS transforms look for pattern like `HEAPxx[...]` and transform it.
+    # This test ensures that one of the transforms doesn't produce a pattern that
+    # another pass can't find anymore, that is that features can work in conjunction.
+    self.do_runf(
+      'other/test_null_deref_via_js.c',
+      emcc_args=args,
+      assert_returncode=NON_ZERO,
+      expected_output=[expected_output])
```

Comment:
Perhaps a single test with all 3 features enabled is enough?

Reply:
It would be enough to reproduce the issue, but I found that having separate tests helps to narrow down which particular transform is at fault.